### PR TITLE
feat: add command to logout of default org

### PR DESCRIPTION
**Pull request by rcoringrato-sfdc**
### What does this PR do?

Add a command `SDFX: Log Out from Default Org` to logout from a projects default org.

### What issues does this PR fix or reference?
@ W-9466522@

### Functionality After

Confirmation dialog when logging out of default scratch org:
ScratchOrgLogoutConfirm3

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.nls.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "force_auth_web_login_authorize_org_text": "SFDX: Authorize an Org",
   "force_auth_access_token_authorize_org_text": "SFDX: Authorize an Org using Session ID",
   "force_auth_logout_all_text": "SFDX: Log Out from All Authorized Orgs",
+  "force_auth_logout_default_text": "SFDX: Log Out from Default Org",
```
Comment (rcoringrato-sfdc) [inline packages/salesforcedx-vscode-core/package.nls.json:7]: @sbudhirajadoc Display title for command as it appears in the command palette.

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts [hunk 2]
```diff
@@ -269,6 +270,11 @@ export const messages = {
   auth_custom_label: 'Custom',
   auth_custom_detail: 'Enter a custom login URL',
   auth_invalid_url: 'URL must begin with http:// or https://',
+  auth_logout_scratch_prompt:
```
Comment (rcoringrato-sfdc) [inline packages/salesforcedx-vscode-core/src/messages/i18n.ts:273]: @sbudhirajadoc Prompt and options for scratch org confirmation dialog.

File: packages/salesforcedx-vscode-core/package.nls.json (hunk 1 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/package.nls.json:7]: Looks good!

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts (hunk 2 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:273]: can I see a screen of this confirmation dialog please?

Comment (rcoringrato-sfdc) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:273]: @sbudhirajadoc I updated the description with an image of the confirmation dialog.

File: packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts [hunk 3]
```diff
@@ -47,3 +58,77 @@ const commandlet = new SfdxCommandlet(
...
+
+export async function forceAuthLogoutDefault() {
+  const { username, isScratch, error } = await resolveDefaultUsername();
+  if (error) {
+    telemetryService.sendException(error.name, error.message);
+    notificationService.showErrorMessage('Logout failed to run');
+  } else if (username) {
+    const logoutCommandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      isScratch
```
Comment (jag-sfdc) [inline packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:97]: Can you please remind why we use a different gatherer for scratchOrgs?

File: packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts [hunk 4]
```diff
@@ -47,3 +58,77 @@ const commandlet = new SfdxCommandlet(
...
+    await logoutCommandlet.run();
+  } else {
+    notificationService.showInformationMessage(
+      nls.localize('auth_logout_no_default_org')
+    );
+  }
+}
+
+async function removeUsername(username: string) {
+  await forceConfigSet('');
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:111]: a general musing - can we eventually replace the call to the `sfdx force:config:set` command with a direct call to `@salesforce/core`?

File: packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts (hunk 3 above)
Comment (rcoringrato-sfdc) [reply packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:97]: @jag-sfdc If you log out of a scratch org without taking prior steps to establish credentials then you will not be able to login again to the org.

File: packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts (hunk 4 above)
Comment (rcoringrato-sfdc) [reply packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:111]: @AnanyaJha Yes, I was planning to convert the spike story into a story to do that. I felt it was out of scope for this work.

File: packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts (hunk 3 above)
Comment (jag-sfdc) [reply packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:97]: Aah, yes. I now remember from the demo discussion.

Comment (jag-sfdc) [reply packages/salesforcedx-vscode-core/src/commands/auth/forceAuthLogout.ts:97]: Can you add this as a comment on why we are doing this.

**Review by jag-sfdc — APPROVED** [review]

**Pull request merged**